src/Entities/Entity.cpp

```diff
@@ -242,10 +242,17 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 {
 	TakeDamageInfo TDI;
 	TDI.DamageType = a_DamageType;
-	TDI.Attacker = a_Attacker;
+	if ((a_Attacker != nullptr) && a_Attacker->IsPawn())
+	{
+		TDI.Attacker = a_Attacker;
+	}
+	else
+	{
+		TDI.Attacker = nullptr;
+	}
```

Comment:
Yes, that would work.

But the whole story is a bit inelegant. 
 1. This will force all users of TDI to come up with such checks, creating dupes. 
 2. Most callers will never care about the projectile entity that hit them and just want to reach the attacker.
 3. The name TDI.Attacker implies an **attacker**.

Can't we break compatibility this one time and fix this? We don't have too many plugins anyways. Fixing now is easier.

If you disagree on breaking compatilbility, I still think TDI should directly contain the shooter for convenience and preventing dupes, under a different variable (e.g. TDI.Shooter). That would at least treat point 1 and 2.